fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -337,6 +340,34 @@ public boolean isLogged() {
         return logged;
     }
 
+    /**
+     * Get the nanosecond time at which this context was opened.
+     * @return time opened
+     */
+    @API(API.Status.INTERNAL)
+    public long getTrackOpenTimeNanos() {
+        return trackOpenTimeNanos;
+    }
+
+    /**
+     * Set the nanosecond time at which this context was opened.
+     * @param trackOpenTimeNanos  time opened
+     */
+    @API(API.Status.INTERNAL)
+    public void setTrackOpenTimeNanos(final long trackOpenTimeNanos) {
```

Comment:
This doesn't feel like it should be `public`. I think I can get behind the others, though they also seem more package-private (unless they're supposed to be visible for testing, or something like that).

Reply:
This one does not need to be public; the getter does, for tests, as you say.